lib/scoped/update.js

```diff
@@ -1,11 +1,57 @@
 module.exports = update
 
 var find = require('./find')
+var merge = require('lodash/merge')
 
 function update (type, api, objectsOrIds, change) {
   return find(type, api, objectsOrIds)
 
   .then(function (storedObjects) {
-    return api.update(objectsOrIds, change)
+    // objectsOrIds may contain updated information for the respective storedObject
+    // (e.g. _deleted = true), so incorporate all information in the objectOrId into
+    // the storedObject
+    var updatedObjects = updateObjects(objectsOrIds, storedObjects)
+
+    return api.update(updatedObjects, change)
+
+    .then(function (apiUpdates) {
+      if (!Array.isArray(apiUpdates)) {
+        return apiUpdates
+      }
+
+      return restoreOriginalFindErrors(updatedObjects, apiUpdates)
+    })
+  })
+}
+
+function updateObjects (objectsOrIds, storedObjects) {
+  if (!Array.isArray(objectsOrIds)) {
+    return updateObject(objectsOrIds, storedObjects)
+  }
+
+  return storedObjects.map(function (storedObject, index) {
+    if (storedObject instanceof Error) {
+      return storedObject
+    }
+
+    return updateObject(objectsOrIds[index], storedObject)
+  })
+}
+
+function updateObject (objectOrId, storedObject) {
+  if (typeof objectOrId !== 'object') {
+    return storedObject
+  }
+
+  return merge(storedObject, objectOrId)
+}
+
+function restoreOriginalFindErrors (updatedObjects, apiUpdates) {
```

Comment:
Note that I have included a `restoreOriginalFindErrors` function, which puts the errors from `find()` back in place of the errors that will be returned from `api.update()`.  For example, if we pass in an array with an out of scope item, `find()` will return this:
```
[
  {
    [Not found: Object with type "test2" and id "F72ECD45-B24A-0E87-A6FC-004C6C7BADA5" is missing]
    name: 'Not found',
    status: 404
  }
]
```
And when this is passed into `api.update()`, then the following error is returned:
```
[
  {
    [Not found: Object with id "undefined" is missing]
    name: 'Not found',
    status: 404
  }
]
```
This function will restore the original out-of-scope error message.  I didn't see any harm in passing the errors through to `api.update()`, as they didn't have an `id` and thus would just return the above error, which we can subsequently replace with the original error.  But please let me know if you prefer if I filter out the errors, then pass the non-errors into `api.update()` and re-insert the original errors, instead.